sphinxcontrib/datatemplates/directive.py

```diff
@@ -28,6 +30,17 @@ def _load_data(self, env, data_source):
         elif data_source.endswith('.json'):
             with open(filename, 'r') as f:
                 return json.load(f)
+        elif data_source.endswith('.csv'):
+            with open(filename, 'r', newline='') as f:
+                sample = f.read(8192)
+                f.seek(0)
+                sniffer = csv.Sniffer()
+                dialect = sniffer.sniff(sample)
+                if 'csvheaders' in self.options:
+                    r = csv.DictReader(f, dialect=dialect)
```

Comment:
I wonder if we want to let the user provide a dialect, too? Or at least some of the parameters that make up a dialect?

Reply:
Added basic dialect support.